pkg/mhooks/event.go

```diff
@@ -2,14 +2,24 @@ package mhooks
 
 import (
 	"encoding/json"
+	"errors"
 	"fmt"
-	"io"
+	"net/http"
 	"time"
 )
 
-func NewEvent(requestBody io.Reader) (*Event, error) {
+func ParseEvent(r *http.Request, secret string) (*Event, error) {
```

Comment:
Originally, I had `CheckSignature` as a separate method that users could call before `ParseEvent`. But this way, we force users to check the signature by just providing the secret. Thoughts?

Reply:
Requiring the signature is checked seems fine to me. People can still unmarshal `Event` if they really want to anyway.